src/archivematica/dashboard/components/accounts/backends.py

```diff
@@ -44,7 +48,35 @@ class CustomOIDCBackend(OIDCAuthenticationBackend):
     Provide OpenID Connect authentication
     """
 
-    def get_settings(self, attr, *args):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        # Store additional settings as instance attributes.
+        self.OIDC_OP_SET_ROLES_FROM_CLAIMS = getattr(
+            settings, "OIDC_OP_SET_ROLES_FROM_CLAIMS", False
+        )
+
+        self.OIDC_OP_ROLE_CLAIM_PATH = getattr(
+            settings, "OIDC_OP_ROLE_CLAIM_PATH", "realm_access.roles"
+        )
+
+        self.OIDC_ACCESS_ATTRIBUTE_MAP = getattr(
+            settings, "OIDC_ACCESS_ATTRIBUTE_MAP", settings.DEFAULT_OIDC_CLAIMS
+        )
+
+        # Valid user roles which may be extracted from OIDC token.
+        self.USER_ROLE_ADMIN = getattr(settings, "USER_ROLE_ADMIN", "admin")
+        self.USER_ROLE_DEFAULT = getattr(settings, "USER_ROLE_DEFAULT", "default")
+
+        # The roles are ordered from highest to lowest permission in this list.
+        # This feature is used in OIDC authentication to determine the highest
+        # permission role of a user based on the claims received when multiple
+        # roles are received.
+        self.USER_ROLES = [
+            (self.USER_ROLE_ADMIN, "Administrator"),
+            (self.USER_ROLE_DEFAULT, "Default"),
+        ]
```

Comment:
Why does this need a two elements tuple if the loop below, which is the only part that accesses it, ignores the second element?

Reply:
I wanted to keep this code the same as in the Storage Service as much as possible. I will update this so that it is a list.

Reply:
I understand there's value in maintaining both implementation in sync. So don't worry about this.